elements/rh-announcement/rh-announcement.ts

```diff
@@ -0,0 +1,130 @@
+import { LitElement, html } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { property } from 'lit/decorators/property.js';
+import { classMap } from 'lit/directives/class-map.js';
+
+import { SlotController } from '@patternfly/pfe-core/controllers/slot-controller.js';
+
+import { colorContextConsumer, type ColorTheme } from '../../lib/context/color/consumer.js';
+import { colorContextProvider } from '../../lib/context/color/provider.js';
+
+import '@rhds/elements/rh-button/rh-button.js';
+
+import styles from './rh-announcement.css';
+
+export class AnnouncementCloseEvent extends Event {
+  constructor() {
+    super('close', { bubbles: true, cancelable: true });
+  }
+}
+
+/**
+ * Announcements are flexible surfaces used to group information in a full width banner layout, traditionally across the top of a page.
+ * They are used to announce new features, promos, or news.
+ * @summary     Arranges content and interactive elements in a layout
+ * @slot        image
+ *              If this slot is used, we expect an image tag with a width and height set.
+ *              An icon, svg, or use of the icon component are also valid in this region.
+ * @slot        Any content that is not designated for the header or footer slot, will go to this slot.
+ * @slot        cta
+ *              If this slot is used, we expect a rh-cta component.
+ * @csspart     container
```

Comment:
there are quite a lot of parts here when i wonder if we really need any of them at all? how customizable does this have to be?

Reply:
This was actually one of the primary concerns for JVP—customizability. He wanted to be sure that they could customize this component as needed.

Reply:
so why not just publish a figma and call it a day?

part is a jackhammer. no one uses the excessive amount of parts and slots in footer. can we decide how much of this we need to support?

I think we need to go over these one-by-one along with design and jvp and decide just how much of this we need to ship